### PR TITLE
Adding time based caching using ehCache, fixes #1

### LINKED ISSUE #1: Caching - We should support a time based cache as well as a size based cache
Basically if the overall cache reaches a certain limit we start purging the old items on a FIFO basis.
Also. if a specific item has been in the cache for more than a configured amount of time and this item is requested again, we refresh the item.

@bjarkih What do you think?

### PR DESCRIPTION
**Pull request by a-flammarion**
Please review Bjarki...

### CODE REVIEW

File: src/main/java/twigkit/frame/CachedImageIOService.java [hunk 1]
```diff
@@ -70,104 +77,184 @@ public void setOfflinePath(String offlinePath) {
...
-				image.setUrl(url);
-
-                if (writeToCache) {
-                    ImageIO.write(image.getBufferedImage(), Image.ContentType.PNG.getSuffix(), file);
-
-                    if (logger.isTraceEnabled()) {
-                        logger.trace("Wrote Image (original) [" + file.getName() + ", " + image.getWidth() + "px by " + image.getHeight() + "px] to cache");
-                    }
+    public Image fromURL(final URL url, boolean writeToCache) throws IOException {
+        Boolean isMissing = false;
```
Comment (bjarkih) [inline src/main/java/twigkit/frame/CachedImageIOService.java]: Use primitive `boolean`.

File: src/main/java/twigkit/frame/CachedImageIOService.java [hunk 2]
```diff
@@ -70,104 +77,184 @@ public void setOfflinePath(String offlinePath) {
...
+        String key = getKeyFromURLBySize(url, 0, 0);
+        if (writeToCache && repository != null && repository.exists() && cache != null && cache.isKeyInCache(key)) {
```
Comment (bjarkih) [inline src/main/java/twigkit/frame/CachedImageIOService.java]: Why are we checking `writeToCache` when reading from the cache? Am I missing something?

File: src/main/resources/cache.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bjarkih) [inline src/main/resources/cache.xml:1]: Did you test adding a custom `/cache.xml` on the class path - I assume that would override this one?

**Comment by bjarkih** [comment]
Just a few comments and questions @a-flammarion.

File: src/main/java/twigkit/frame/CachedImageIOService.java (hunk 2 above)
Comment (a-flammarion) [inline src/main/java/twigkit/frame/CachedImageIOService.java]: as far as I can see, `writeToCache` is used as a flag to read and write from the cache. aybe it should be renamed to `useCache` ?

File: src/main/resources/cache.xml (hunk 3 above)
Comment (a-flammarion) [inline src/main/resources/cache.xml:1]: I haven't. How do I set the class path when I run an app using `mvn jetty:run` ?

Comment (bjarkih) [inline src/main/resources/cache.xml:1]: You can put that in `src/main/resources`.

File: src/main/java/twigkit/frame/CachedImageIOService.java (hunk 2 above)
Comment (bjarkih) [inline src/main/java/twigkit/frame/CachedImageIOService.java]: Yes, `useCache` is more descriptive.

**Pull request merged**